### PR TITLE
Full reverse-proxy functionality

### PR DESCRIPTION
**Pull request by KyCodeHuynh**
This adds the remaining reverse-proxy functionality (302 redirects + relative link rewriting), as well as unit + integration tests.

### CODE REVIEW

File: reverse_proxy_handler.h [hunk 1]
```diff
@@ -9,9 +9,17 @@
 // to which it actually reaches out.
 //
 // See: https://github.com/UCLA-CS130/Mr.-Robot-et-al./issues/62
+
+// We use this in reverse_proxy_hanlder_test.cc
```
Comment (gotitmemorize) [inline reverse_proxy_handler.h]: Typo here

File: reverse_proxy_handler.cc [hunk 2]
```diff
@@ -51,88 +52,61 @@ RequestHandler::Status ReverseProxyHandler::Init(const std::string& uri_prefix,
...
+// Helper for parsing response code from origin responses
+std::string ReverseProxyHandler::getRemoteResponseCode(const std::string& response_buffer_string) {
     // Response has the form:
     // HTTP/1.0 200 OK
     // Content-Length: 3497
     // Content-Type: text/plain
+    //
     // <body contents>
-
+    //
```
Comment (gotitmemorize) [inline reverse_proxy_handler.cc]: Remove empty comment

File: reverse_proxy_handler.h [hunk 3]
```diff
@@ -3,10 +3,23 @@
...
+class ReverseProxyHandlerTest;
+
 // Retrieves a file from the server's file system
 class ReverseProxyHandler : public RequestHandler {
 public:
 
+    // We use friend so that ReverseProxyHandlerTest can access private variables for testing
+    friend class ReverseProxyHandlerTest;
```
Comment (gotitmemorize) [inline reverse_proxy_handler.h]: That's a good idea, I like it!

File: reverse_proxy_handler.h [hunk 4]
```diff
@@ -3,10 +3,23 @@
...
 #include "request_handler.h"
 
+// Interacts with clients from the Internet at-large
+// as if this server was the one serving up the
+// content requested, rather than the remote servers
+// to which it actually reaches out.
+//
+// See: https://github.com/UCLA-CS130/Mr.-Robot-et-al./issues/62
+
+// We use this in reverse_proxy_hanlder_test.cc
```
Comment (gotitmemorize) [inline reverse_proxy_handler.h]: Typo in comment

File: reverse_proxy_handler.cc (hunk 2 above)
Comment (KyCodeHuynh) [inline reverse_proxy_handler.cc]: This is actually a separator line, which is non-obvious from the diff: 

image

File: reverse_proxy_handler.h (hunk 3 above)
Comment (KyCodeHuynh) [inline reverse_proxy_handler.h]: All credit to @kfrankc

**Pull request merged**